### PR TITLE
Add wordcount.js integration test

### PR DESCRIPTION
**Pull request by lmcnulty**
Fulfilling [#607](https://github.com/responsible-ai-collaborative/aiid/issues/607), tests that:
- table and word cloud both exist 
- words in table have length > 2 and count > 10.

### CODE REVIEW

File: site/gatsby-site/cypress/integration/wordcounts.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+describe('The Word Counts Page', () => {
+  it('successfully loads', () => {
+    cy.visit('/summaries/wordcounts');
+  });
+
+  it('word count table should exist', () => {
+    cy.visit('/summaries/wordcounts');
```
Comment (smcgregor) [inline site/gatsby-site/cypress/integration/wordcounts.js:7]: These CSS-based selectors should detect whether the table as it currently exists is present or not, but if/when the display of the table changes this test will naturally fail. I think this test as a basic "did it load" test would be fine just checking for the header text of `Word Counts`. Your next test, which looks at the contents of the table, is then looking at the more specific question of the contents of the table.

File: site/gatsby-site/cypress/integration/wordcounts.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      })
+      .then(() => {
+        const ascending = (a, b) => a - b;
+
+        expect(words.map((word) => word.length).sort(ascending)[0] > 2).to.be.true;
+        expect(counts.sort(ascending)[0] > 9).to.be.true;
+      });
+  });
+  it('word cloud should exist', () => {
+    cy.get('.col-8 svg');
```
Comment (smcgregor) [inline site/gatsby-site/cypress/integration/wordcounts.js:44]: These CSS selectors are likely an indication that the word clouds should have a handle that can be checked for that is not related to the templating. Maybe a CSS class that has no associated styling (at present) would be good to add?

**Review by smcgregor — COMMENTED** [review]
Two inline comments. @cesarvarela do you agree with my comments? Please feel free to review this one as you have written a lot more Cypress than I have.

File: site/gatsby-site/cypress/integration/wordcounts.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+describe('The Word Counts Page', () => {
+  it('successfully loads', () => {
+    cy.visit('/summaries/wordcounts');
+  });
+
+  it('word count table should exist', () => {
+    cy.visit('/summaries/wordcounts');
+    cy.get('.row .list-group');
+    cy.get('.row .list-group > div');
```
Comment (cesarvarela) [inline site/gatsby-site/cypress/integration/wordcounts.js:9]: I agree with the comment above. Also, this seems to be missing assertions? It is getting the DOM nodes but not doing anything with them:

Something like `cy.get('.row .list-group').should('exist');`

Comment (cesarvarela) [reply site/gatsby-site/cypress/integration/wordcounts.js:9]: https://docs.cypress.io/api/commands/should

File: site/gatsby-site/cypress/integration/wordcounts.js (hunk 2 above)
Comment (cesarvarela) [reply site/gatsby-site/cypress/integration/wordcounts.js:44]: Yes, cypress recommends using the `data-cy="something"` attribute to keep tests separate from CSS. It is not expected to add this to every component, but only the top ones.

https://docs.cypress.io/guides/references/best-practices#Selecting-Elements

File: site/gatsby-site/cypress/integration/wordcounts.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        let count = 0;
+
+        let countText = row.find('span').text().trim();
+
+        let spanIsNumeric = /\d+/.test(countText);
+
+        if (spanIsNumeric) {
+          count = parseInt(countText);
+        }
+        counts.push(count);
```
Comment (cesarvarela) [inline site/gatsby-site/cypress/integration/wordcounts.js:34]: Looks weird to me to have this much logic in a test, but maybe there is no better way to do it. Ideally, it should be easier to tell what the test it is doing.

nitpick: variables that don't change should be declared as `const` instead of `let`.

File: site/gatsby-site/cypress/integration/wordcounts.js (hunk 3 above)
Comment (cesarvarela) [reply site/gatsby-site/cypress/integration/wordcounts.js:9]: Yeah, the test will fail, but leaving it like that, it is not clear what you are testing. Because that test will pass no matter how many items there are or even if they are visible or not, as long as there is at least one present on the DOM, but that's a little bit open-ended.

You probably want something like `cy.get('.row .list-group').should('have.length', 2).and('be.visible')` (didn't test it).

**Review by smcgregor — APPROVED** [review]
Approved, but @cesarvarela is the strongest in Cypress so I will let him do a final review.

**Comment by smcgregor** [comment]
@lmcnulty feel free to resolve review discussion points in the event you believe it likely that the person opening the discussion is satisfied with the changes you introduced to address the discussion.

**Review by cesarvarela — APPROVED** [review]
